nucypher/blockchain/eth/agents.py

```diff
@@ -1027,6 +1027,25 @@ def cancel_bid(self, checksum_address: str) -> dict:
         receipt = self.blockchain.send_transaction(contract_function=contract_function, sender_address=checksum_address)
         return receipt
 
+    def force_refund(self, checksum_address: str, addresses: List[str]) -> dict:
+        """Force refund to bidders who can get tokens more than maximum allowed."""
+        addresses = sorted(addresses, key=str.casefold)
+        contract_function = self.contract.functions.forceRefund(addresses)
+        receipt = self.blockchain.send_transaction(contract_function=contract_function, sender_address=checksum_address)
+        return receipt
+
+    @validate_checksum_address
+    def verify_bidding_correctness(self,
+                                   checksum_address: str,
+                                   gas_limit: int,  # TODO - #842: Gas Management
+                                   gas_to_save_state: int = 30000) -> dict:
```

Comment:
Can you remind me how `30000` became the default here?

Reply:
It's safe amount to save state in verrify method in contract:
```
        if (index != nextBidderToCheck) {
            emit BiddersChecked(msg.sender, nextBidderToCheck, index);
            nextBidderToCheck = index;
        }
```
Value depends on gas costs for saving state and emitting event

Reply:
As a matter of admin UX - How can these values be determined for the user?

Reply:
only try to run with different values